### ISSUE TITLE
Glitchy ListBox when adding items & setting SelectedIndex

### ISSUE THREAD

**Issue by metal450** [AvaloniaUI/Avalonia#16014]
### Describe the bug

In my app, I've got a page where I show Log messages in a ListBox:

```
<ListBox ItemsSource="{Binding LogItems}" SelectedIndex="{Binding SelectedIndex}">
  <ListBox.ItemTemplate>
      <DataTemplate>
         <Border BorderThickness="0,0,0,1" Padding="5 0" >
             <TextBlock Text="{Binding}" Margin="0" Padding="0" TextWrapping="Wrap"/>
         </Border>
     </DataTemplate>
  </ListBox.ItemTemplate>
</ListBox>
```
And:
```
    public ObservableCollection<string> LogItems
    {
        get => _logItems;
        set => this.RaiseAndSetIfChanged(ref _logItems, value);
    }
    private ObservableCollection<string> _logItems = new ObservableCollection<string>();

    public int SelectedIndex
    {
        get => _selectedIndex;
        set => this.RaiseAndSetIfChanged(ref _selectedIndex, value);
    }
    private int _selectedIndex = -1;

    public void AddLog(string log)
    {
        LogItems.Add(log);
        //SelectedIndex = LogItems.Count - 1;
    }
```
The above code works fine.

However, if I uncomment the SelectedItem line so that the ListBox will scroll to the bottom each time a new log is added, sometimes it works fine, but randomly it gets all glitched out, with multiple messages overlapping each other. Once that happens, if I scroll up, those overlapping messages will even jump around at random. Here's a video to illustrate:

![peek](https://github.com/AvaloniaUI/Avalonia/assets/1877409/3c5497ac-2dfa-417a-b228-106ed202a7b7)

The behavior is reproducible on Windows, Android, and iOS. It happens on the latest stable version of Avalonia, 11.0.10.

### To Reproduce

See description

### Avalonia version

11.0.10

**Comment by timunie** [comment]
please try 11.1 beta as well. Also looks like a custom theme, so best to check with simple sample and vanilla Avalonia

**Comment by jp2masa** [comment]
Duplicate of #14389?

**Comment by metal450** [comment]
> please try 11.1 beta as well.

Hmm, actually seems to be fixed in the beta. Is the beta safe to use in a production app tho...?

**Comment by timunie** [comment]
I'd use it as we are close to 11.1 release. But better to test your App carefully before releasing it.

**Comment by metal450** [comment]
Unfortunately I haven't been able to get 11.1 to work in practice, because it requires .NET 8 (currently I'm on .net 7), and when I upgrade to .net 8, I can no longer initialize Maui.Essentials on iOS.

I'm assuming there's no way to either:
* Get this fix without upgrading to Avalonia 11.1, or
* Upgrade to 11.1 without having to upgrade to .NET 8, or
* I posted the Maui.Essentials issue [here](https://github.com/dotnet/maui/issues/23145); have you seen this occur when upgrading to .Net 8? I know it isn't an Avalonia issue...but I'm kind of stuck if Avalonia is now requiring .Net 8. I included a repro project.
